internal/client/k8s_client.go

```diff
@@ -0,0 +1,198 @@
+type K8sClient struct {
+	cfg       *rest.Config
+	disco     discovery.DiscoveryInterface
+	dyn       dynamic.Interface
+	mapper    meta.RESTMapper
+	Namespace string // optional default for namespaced resources
+}
+
+// NewK8sFromRESTConfig constructs a fully-initialized client (no kubeconfig fallback).
+func NewK8sFromRESTConfig(cfg *rest.Config) (*K8sClient, error) {
+	disco, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("discovery: %w", err)
+	}
+	dyn, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("dynamic: %w", err)
+	}
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(disco))
+	return &K8sClient{cfg: cfg, disco: disco, dyn: dyn, mapper: mapper}, nil
+}
+
+// Preflight verifies the API server is reachable for the selected context.
+func (c *K8sClient) Preflight(ctx context.Context) error {
+	if _, err := c.disco.ServerVersion(); err != nil {
+		return fmt.Errorf("kubernetes API unreachable: %w", err)
+	}
+	return nil
+}
+
+// ListCRDs returns native CRDs (the MCP layer shapes any view structs).
+func (c *K8sClient) ListCRDs(ctx context.Context) ([]*apiextv1.CustomResourceDefinition, error) {
+	cs, err := apiextcs.NewForConfig(c.cfg)
+	if err != nil {
+		return nil, fmt.Errorf("apiextensions client: %w", err)
+	}
+	list, err := cs.ApiextensionsV1().CustomResourceDefinitions().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	items := make([]*apiextv1.CustomResourceDefinition, 0, len(list.Items))
+	for i := range list.Items {
+		crd := list.Items[i]
+		items = append(items, &crd)
+	}
+	return items, nil
+}
+
+// GetCRD fetches a CRD by name and renders it as yaml|json|describe (default yaml).
+func (c *K8sClient) GetCRD(ctx context.Context, name, mode string) (string, error) {
+	cs, err := apiextcs.NewForConfig(c.cfg)
+	if err != nil {
+		return "", fmt.Errorf("apiextensions client: %w", err)
+	}
+	crd, err := cs.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	switch strings.ToLower(mode) {
+	case "json":
+		b, err := json.MarshalIndent(crd, "", "  ")
+		if err != nil {
+			return "", err
+		}
+		return string(b), nil
+	case "describe":
+		var sb strings.Builder
+		fmt.Fprintf(&sb, "Name: %s\n", crd.Name)
+		fmt.Fprintf(&sb, "Group: %s\n", crd.Spec.Group)
+		fmt.Fprintf(&sb, "Kind: %s\n", crd.Spec.Names.Kind)
+		fmt.Fprintf(&sb, "Scope: %s\n", crd.Spec.Scope)
+		versions := make([]string, 0, len(crd.Spec.Versions))
+		for _, v := range crd.Spec.Versions {
+			versions = append(versions, v.Name)
+		}
+		fmt.Fprintf(&sb, "Versions: %s\n", strings.Join(versions, ", "))
+		return sb.String(), nil
+	default: // yaml
+		b, err := json.Marshal(crd)
+		if err != nil {
+			return "", err
+		}
+		y, err := syaml.JSONToYAML(b)
+		if err != nil {
+			return "", err
+		}
+		return string(y), nil
+	}
+}
```

Comment:
This method could probably be moved to the MCP server package since it's specific to rendering a view for the MCP server. The `output` package has been where we've been building out core functions for rendering resources, but that may not be applicable in this case.